### ISSUE TITLE
Header failing online accessibility checker

### ISSUE THREAD

**Issue by 72gm** [scottish-government-design-system/design-system#29]
Hi,

When checking the accessibility of my app, using the online tools, the header was failing. Checking the SH header component on the SG website, it also fails e.g.

![image](https://github.com/scottishgovernment/pattern-library/assets/12847881/d7519293-0d35-4942-9c7c-7dced32cfb59)

Thanks

**Comment by jsutcliffe** [comment]
This should be fixed as of version 0.0.349.

**Comment by 72gm** [comment]
I've updated but appear to still be getting the same issue? My node modules also has the correct 349 version

![image](https://github.com/scottishgovernment/pattern-library/assets/12847881/e76e8890-cba1-42bb-a3cd-1034d37c998a)

**Comment by jsutcliffe** [comment]
Looking at your screenshot it looks like you've using the latest version. You should see a "Menu" label that controls the menu checkbox still in place next to the "Menu" button when styles are turned off (the fix to your original issue makes it so that that label is now hidden with display:none instead of removed from the DOM). Do you need to rebuild your scripts, or clear a cache?

<img width="183" alt="Screen Shot 2023-05-25 at 16 03 05" src="https://github.com/scottishgovernment/pattern-library/assets/369528/da1d1d3c-258e-403b-ad8c-4737dcf2a507">

**Comment by 72gm** [comment]
Yeah as you said, the issue was resolved with an update to the html (update of button to a label)... as opposed to the CSS ( it is only the CSS that we use dynamically) .. so in our React header component this fix didn't replicate... 

To fix this we had to update the html in our React component with the new header html.. which sort identifies a flaw for people who have generated React/Angular/Vue components from the pattern library 

i.e. when do we know that we should update our html as we won't necessarily know if there has been an update to the html (and it isn't a straight like for like copy as we will have optimised it for our framework)

**Comment by 72gm** [comment]
Actually I'll need to look into this

The code that is in the sample html on your website fixes that issue but actually breaks the keyboard navigation on mobile... has the js changed as well?

The sample code is also different from what is running on the website...

**Comment by jsutcliffe** [comment]
There were no changes to the base HTML, only to some DOM transformations that occur when the component's JavaScript is initialised (the labels are now given a CSS class to hide them, instead of being removed from the DOM).

One way we try to avoid breaking changes at the moment is to avoid making changes to the base HTML unless it is truly unavoidable. Modifications to published components are almost always restricted to CSS and/or JavaScript, so that users only need to update their styles and scripts to get any improvements we make to components, rather than needing to also modify their templates.

As we are a small team with no in-house React experience it is difficult to offer support to people implementing the Design System in a React context, but perhaps something you could try is basing your React components' HTML on the DS components' base HTML (i.e. what we provide in the "sample HTML" blocks for each component) instead of the components' HTML post-initialisation, and let the Design System JavaScript carry out the DOM transformations.

**Comment by 72gm** [comment]
Yeah we normally use the sample html code as our baseline. 

I imagine, in this instance, that perhaps we decided to use the transformed code as it was easier for writing our own JS ( as we can't get your JS to work for previously advised reasons) 

It looks like the change we are interested in is in 348 so we will try implementing this in our component library
